build: Makefile treats msys as Windows+powershell

### Problem

The newest Makefile gives this:

    E:\neovim_sources>make -v
    GNU Make 4.4.1
    Built for x86_64-pc-msys
    Copyright (C) 1988-2023 Free Software Foundation, Inc.
    License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
    This is free software: you are free to change and redistribute it.
    There is NO WARRANTY, to the extent permitted by law.
    
    E:\neovim_sources>cmake --version
    cmake version 3.30.5
    
    CMake suite maintained and supported by Kitware (kitware.com/cmake).

    E:\neovim_sources>make distclean
    if (Test-Path  ".deps") { remove-item -recurse  ".deps" }
    if (Test-Path  build) { remove-item -recurse  build }
    make clean
    make[1]: Entering directory '/e/neovim_sources'
    /usr/bin/make -C test/old/testdir clean
    make[2]: Entering directory '/e/neovim_sources/test/old/testdir'
    rm -f -rf *.out *.failed *.res *.rej *.orig *.tlog opt_test.vim test_result.log test.log messages starttime test.out X* viminfo test.ok valgrind.* .*.swp .*.swo .gdbinit /e/neovim_sources/test/old/testdir/X-test-tmpdir del ../../../runtime/doc/.*.swp
    make[2]: Leaving directory '/e/neovim_sources/test/old/testdir'
    /usr/bin/make -C runtime/indent clean
    make[2]: Entering directory '/e/neovim_sources/runtime/indent'
    rm -f testdir/*.fail testdir/*.out
    make[2]: Leaving directory '/e/neovim_sources/runtime/indent'
    make[1]: Leaving directory '/e/neovim_sources'


    E:\neovim_sources>make CMAKE_BUILD_TYPE=Release CMAKE_INSTALL_PREFIX=D:/nvim install
    At line:1 char:3
    + if [ -f build/.ran-cmake ]; then \
    +   ~
    Missing '(' after 'if' in if statement.
    At line:1 char:5
    + if [ -f build/.ran-cmake ]; then \
    +     ~
    Missing type name after '['.
    At line:2 char:57
    +   cached_prefix=At line:1 char:21 + cmake -L -N build | 2>/dev/null g ...
    +                                                         ~
    Expressions are only allowed as the first element of a pipeline.
    At line:2 char:69
    + ... efix=At line:1 char:21 + cmake -L -N build | 2>/dev/null grep 'CMAKE_ ...
    +                                                              ~~~~
    Unexpected token 'grep' in expression or statement.
    At line:2 char:240
    + ... ment of a pipeline. At line:1 char:33 + cmake -L -N build | 2>/dev/nu ...
    +                                                                 ~
    Expressions are only allowed as the first element of a pipeline.
    At line:2 char:252
    + ... ine. At line:1 char:33 + cmake -L -N build | 2>/dev/null grep 'CMAKE_ ...
    +                                                              ~~~~
    Unexpected token 'grep' in expression or statement.
    At line:3 char:5
    +   if ! [ "D:/nvim" = "$cached_prefix" ]; then \
    +     ~
    Missing '(' after 'if' in if statement.
    At line:3 char:9
    +   if ! [ "D:/nvim" = "$cached_prefix" ]; then \
    +         ~
    Missing type name after '['.
    At line:3 char:7
    +   if ! [ "D:/nvim" = "$cached_prefix" ]; then \
    +       ~
    Missing expression after unary operator '!'.
    At line:3 char:8
    +   if ! [ "D:/nvim" = "$cached_prefix" ]; then \
    +        ~
    Unexpected token '[' in expression or statement.
        + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
        + FullyQualifiedErrorId : MissingOpenParenthesisInIfStatement
    
    make: *** [Makefile:50: checkprefix] Error 1
`

### Steps to reproduce

        
        make CMAKE_BUILD_TYPE=Release CMAKE_INSTALL_PREFIX=D:/nvim install

### Expected behavior

no error message and make succeeds.

### Nvim version (nvim -v)

nvim v0.11.0-dev-649+ge48179f31-dirty

### Vim (not Nvim) behaves the same?

no

### Operating system/version

windows 10  22H2 (19045. 5073)

### Terminal name/version

cmd/powershell 7

### $TERM environment variable

-

### Installation

-

Here's the output of SET command: 

    E:\neovim_sources>set
    ALLUSERSPROFILE=C:\ProgramData
    APPDATA=C:\Users\alex\AppData\Roaming
    CC=clang
    CommonProgramFiles=C:\Program Files\Common Files
    CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
    CommonProgramW6432=C:\Program Files\Common Files
    COMPUTERNAME=WIN10
    ComSpec=C:\Windows\system32\cmd.exe
    DriverData=C:\Windows\System32\Drivers\DriverData
    HOMEDRIVE=C:
    HOMEPATH=\Users\alex
    LOCALAPPDATA=C:\Users\alex\AppData\Local
    LOGONSERVER=\\WIN10
    NUMBER_OF_PROCESSORS=4
    OS=Windows_NT
    Path=C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\PowerShell\7;C:\Program Files\PowerShell\7\;E:\msys64\mingw64\bin;e:\msys64\usr\bin;d:\nvim\bin;E:\lua-language-server\bin;C:\Users\alex\AppData\Local\Microsoft\WindowsApps;
    PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
    POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows 10 Pro
    PROCESSOR_ARCHITECTURE=AMD64
    PROCESSOR_IDENTIFIER=AMD64 Family 21 Model 48 Stepping 1, AuthenticAMD
    PROCESSOR_LEVEL=21
    PROCESSOR_REVISION=3001
    ProgramData=C:\ProgramData
    ProgramFiles=C:\Program Files
    ProgramFiles(x86)=C:\Program Files (x86)
    ProgramW6432=C:\Program Files
    PROMPT=$P$G
    PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules
    PUBLIC=C:\Users\Public
    SESSIONNAME=Console
    SETUPTOOLS_USE_DISTUTILS=stdlib
    SystemDrive=C:
    SystemRoot=C:\Windows
    TEMP=f:\tmp
    TMP=f:\tmp
    USERDOMAIN=WIN10
    USERDOMAIN_ROAMINGPROFILE=WIN10
    USERNAME=alex
    USERPROFILE=C:\Users\alex
    windir=C:\Windows
    WSLENV=WT_SESSION:WT_PROFILE_ID:
    
I didn't see anything relevant here, except for msys/mingw in the PATH.

Since `checkprefix` is in error msg, so I tried to build without the `CMAKE_INSTALL_PREFIX` and got this:
    
    E:\neovim_sources>make CMAKE_BUILD_TYPE=Release install    
    cmake -S /e/neovim_sources//cmake.deps -B ".deps" -G Ninja
    CMake Error: The source directory "/e/neovim_sources/cmake.deps" does not exist.
    Specify --help for usage, or press the help button on the CMake GUI.
    make: *** [Makefile:108: build/.ran-deps-cmake] Error 1
    
    E:\neovim_sources>
    E:\neovim_sources>cd cmake.deps
    
    E:\neovim_sources\cmake.deps>

It looks like the errors narrowed down to the directories difference between cmd and bash.

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,14 @@
 ifeq ($(OS),Windows_NT)
+  ifeq '$(findstring ;,$(PATH))' ';'
+    UNIX_LIKE := FALSE
+  else
+    UNIX_LIKE := TRUE
+  endif
+else
+  UNIX_LIKE := TRUE
+endif
+
+ifeq ($(UNIX_LIKE),FALSE)
   SHELL := powershell.exe
   .SHELLFLAGS := -NoProfile -NoLogo
   MKDIR := @$$null = new-item -itemtype directory -force

